### ISSUE TITLE
corrupt stream with extra bytes is not rejected

### ISSUE THREAD

**Issue by jsha** [dropbox/rust-brotli#83]
Steps to reproduce:

Read a Brotli stream with extra junk at the end.

Expected result:

Receive an Err from brotli_decompressor::Decompressor::read.

Actual result:

No error.

For comparison, the `brotli` command line tool rejects streams with extra junk at the end:

```
$ (brotli -c <<<"hello"; echo goodbye) > corrupt.br
$ brotli -d < corrupt.br
hello
corrupt input [con]
$ echo $?
1
```

Test code to reproduce:

```
    #[test]
    fn test_no_leftovers() {
        // Output from this command:
        // (brotli -c <<<"hello"; echo goodbye) | xxd -p | sed 's/../\\x&/g'
        let compressed_with_extra = b"\x8f\x02\x80\x68\x65\x6c\x6c\x6f\x0a\x03\x67\x6f\x6f\x64\x62\x79\x65\x0a";
        let mut reader = brotli_decompressor::Decompressor::new(Cursor::new(compressed_with_extra), 8000);
        assert!(std::io::read_to_string(&mut reader).is_err());
    }
```

**Comment by danielrh** [comment]
Fixed by 0045b6e18a91b12937ffc45a7a4f934ffeab7d2e in rust-brotli-decompressor

**Comment by catenacyber** [comment]
Looks like the same root cause as https://github.com/dropbox/rust-brotli-decompressor/issues/24
